Maple2.Server.Game/PacketHandlers/ItemDismantleHandler.cs

```diff
@@ -173,10 +174,28 @@ private enum Command : byte {
 
             while (i < items.Count) {
                 Item item = items[i++];
+                                
                 if (session.DismantleStaging.Any(existing => existing.Uid == item.Uid)) {
                     continue;
                 }
 
+                if (item.IsLocked) {
+                    continue;
+                }
+
+                switch (item.Inventory.ToString()) {
+                    case "Gear" or "Gemstone":
+                        if (!item.Metadata.Limit.EnableBreak) {
+                            continue;
+                        }
+                        break;
+                    case "Outfit":
+                        if (item.GachaDismantleId == 0) {
+                            continue;
+                        }
+                        break;
+                }
```

Comment:
i'm not totally a fan of `ToString`-ing an enum. It allows for vulnerabilities if we ever change the enum names. You can still do this switch but just by the enum instead.